packages/components/src/confirm/component.tsx

```diff
@@ -0,0 +1,80 @@
+// @ts-nocheck (for now)
+
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import { Ref, useEffect } from 'react';
+import { confirmable } from 'react-confirm';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import type { OwnProps } from './types';
+import { Card, CardHeader, CardFooter } from '../card';
+import { Heading } from '../heading';
+import { contextConnect, PolymorphicComponentProps } from '../ui/context';
+import { useConfirm } from './hook';
+import type { KeyboardEvent } from 'react';
+
+// @todo add type declarations for the react-confirm functions
+function Confirm(
+	props: PolymorphicComponentProps< OwnProps, 'div', false >,
+	forwardedRef: Ref< any >
+) {
+	const {
+		role,
+		wrapperClassName,
+		overlayClassName,
+		show,
+		proceed,
+		confirmation,
+		...otherProps
+	} = useConfirm( props );
+
+	function handleEscapePress( event: KeyboardEvent< HTMLDivElement > ) {
+		// `keyCode` is deprecated, so let's use `key`
+		if ( event.key === 'Escape' ) {
+			proceed( false );
+		}
+	}
+
+	useEffect( () => {
+		document.addEventListener( 'keydown', handleEscapePress );
+		return () =>
+			document.removeEventListener( 'keydown', handleEscapePress );
+	} );
+
+	return (
+		<div
+			role={ role }
+			ref={ forwardedRef }
+			className={ wrapperClassName }
+			style={ { visibility: show ? 'visible' : 'hidden' } }
```

Comment:
Not sure if this will still apply if we re-use an existing `Modal` component, but all styles should be applied via Emotion.

In this specific case, you could export a variable from `styles.js` called

```
const wrapperHidden = css`
  visibility: hidden
`;
```

I left another comment in the `useConfirm` hook to show how you would then conditionally apply those styles to the wrapper.

Reply:
Thanks! See my reply here: https://github.com/WordPress/gutenberg/pull/34153#discussion_r695359938.